src/mlpack/tests/binarize_test.cpp

```diff
@@ -57,15 +55,13 @@ BOOST_AUTO_TEST_CASE(BinerizeAll)
 
   Binarize<double>(input, output, threshold);
 
-  BOOST_REQUIRE_SMALL(output(0, 0), 1e-5); // 1
-  BOOST_REQUIRE_SMALL(output(0, 1), 1e-5); // 2
-  BOOST_REQUIRE_SMALL(output(0, 2), 1e-5); // 3
-  BOOST_REQUIRE_SMALL(output(1, 0), 1e-5); // 4
-  BOOST_REQUIRE_SMALL(output(1, 1), 1e-5); // 5
-  BOOST_REQUIRE_CLOSE(output(1, 2), 1.0, 1e-5); // 6
-  BOOST_REQUIRE_CLOSE(output(2, 0), 1.0, 1e-5); // 7
-  BOOST_REQUIRE_CLOSE(output(2, 1), 1.0, 1e-5); // 8
-  BOOST_REQUIRE_CLOSE(output(2, 2), 1.0, 1e-5); // 9
+  REQUIRE(output(0, 0) == Approx(0.0).margin(1e-5)); // 1
+  REQUIRE(output(0, 1) == Approx(0.0).margin(1e-5)); // 2
+  REQUIRE(output(0, 2) == Approx(0.0).margin(1e-5)); // 3
+  REQUIRE(output(1, 0) == Approx(0.0).margin(1e-5)); // 4
+  REQUIRE(output(1, 1) == Approx(0.0).margin(1e-5)); // 5
+  REQUIRE(output(1, 2)== Approx(1.0).epsilon(1e-7)); // 6
+  REQUIRE(output(2, 0)== Approx(1.0).epsilon(1e-7)); // 7
+  REQUIRE(output(2, 1)== Approx(1.0).epsilon(1e-7)); // 8
+  REQUIRE(output(2, 2)== Approx(1.0).epsilon(1e-7)); // 9
```

Comment:
```suggestion
  REQUIRE(output(1, 2) == Approx(1.0).epsilon(1e-7)); // 6
  REQUIRE(output(2, 0) == Approx(1.0).epsilon(1e-7)); // 7
  REQUIRE(output(2, 1) == Approx(1.0).epsilon(1e-7)); // 8
  REQUIRE(output(2, 2) == Approx(1.0).epsilon(1e-7)); // 9
```
Some more simple style fixes. :)